core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -336,7 +336,7 @@ public void handleInStoreExpiration(K key) {
          // Unfortunately stores don't pull the entry so we can't tell exactly why it expired and thus we have to remove
          // the entire value.  Unfortunately this could cause a concurrent write to be undone
          try {
-            cache.removeLifespanExpired(key, null, null).join();
+            cache.withFlags(Flag.SKIP_SHARED_CACHE_STORE).removeLifespanExpired(key, null, null).join();
```

Comment:
Technically we need to do this with the other `handleInStoreExpiration` method. But I will be fixing this with my upcoming expiration changes (just have a couple core tests failures that are alluding me as to how to fix yet).

Reply:
Oops, both of us missed the other method! Let me open a PR real quick :)

Reply:
It isn't high priority. The other method is only used with stores that implement the `AdvancedCacheExpirationWriter` interface of which is currently only the `DummyInMemoryStore`.

Reply:
I opened https://github.com/infinispan/infinispan/pull/7704 anyway :)